paymentsheet/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/bacs/BacsMandateConfirmationForm.kt

```diff
@@ -0,0 +1,239 @@
+package com.stripe.android.paymentsheet.paymentdatacollection.bacs
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.google.android.gms.common.util.VisibleForTesting
+import com.stripe.android.core.strings.resolvableString
+import com.stripe.android.paymentsheet.R
+import com.stripe.android.paymentsheet.utils.PaymentSheetContentPadding
+import com.stripe.android.uicore.StripeTheme
+import com.stripe.android.uicore.strings.resolve
+import com.stripe.android.uicore.stripeColors
+import com.stripe.android.uicore.text.Html
+
+@Composable
+internal fun BacsMandateConfirmationFormScreen(
+    viewModel: BacsMandateConfirmationViewModel = viewModel()
+) {
+    val viewState by viewModel.viewState.collectAsState()
+
+    BacsMandateConfirmationFormView(viewState, viewModel::handleViewAction)
+}
+
+@Composable
+@VisibleForTesting
+internal fun BacsMandateConfirmationFormView(
```

Comment:
Can we add paparazzi tests for this? I see one, typically we add one variant to ensure layout is correct at the top level, and individual tests for the subcomponents to ensure they are correct for each variant.